web-common/src/features/dashboards/state-managers/loaders/test/PageMock.ts

```diff
@@ -0,0 +1,87 @@
+import type { afterNavigate } from "$app/navigation";
+import type { AfterNavigate, Page } from "@sveltejs/kit";
+import { writable, get, type Readable, type Updater } from "svelte/store";
+import { expect } from "vitest";
+
+export type HoistedPage = Readable<Page> & {
+  goto: (path: string, opts?: { replaceState?: boolean }) => void;
+  afterNavigate: typeof afterNavigate;
+};
+
+export class PageMock {
+  private readonly update: (updater: Updater<Page>) => void;
+  private afterNavigateCallback: (navigation: AfterNavigate) => void;
+  // Save the url search history to assert that extra entries are not added.
+  public urlSearchHistory: string[] = [];
+
+  public constructor(private readonly hoistedPage: HoistedPage) {
+    const { update, subscribe } = writable<Page>({
+      url: new URL("http://localhost/explore/AdBids_explore"),
```

Comment:
This hardcoded URL path makes this class a little hard to follow. I was confused when I saw the `gotoSearch` method rather than a simpler `goto`, then I realized it was because the base URL path was already assumed